zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/Interpreter.java

```diff
@@ -53,6 +54,24 @@
  */
 public abstract class Interpreter {
 
+  // Flag for creating intp-process in a cluster
+  // 1. This flag is true in cluster mode, otherwise false
+  // 2. If this flag is false, then the intp-process will be created locally.
+  // 3. If this flag is true then the intp-process will be created in the cluster
+  // 4. When zepplin-server A notifies zepplin-server B to create intp process,
+  //    this flag in zepplin-server B is false, and zepplin-server B creates intp process locally.
+  protected boolean createInptProcessInCluster = false;
+
+  public void setCreateInptProcessInCluster(boolean createInptProcessInCluster) {
```

Comment:
I don't think it is proper to put interpreter launch logic in Interpreter itself, it should be the interpreter launch component to handle it.

Reply:
Ok, accept your opinion, Let me solve this problem again.